runtime/src/main/cpp/ToString.cpp

```diff
@@ -108,4 +108,16 @@ OBJ_GETTER(Kotlin_Long_toStringRadix, KLong value, KInt radix) {
   RETURN_RESULT_OF(Kotlin_toStringRadix<KLong>, value, radix)
 }
 
+OBJ_GETTER(Kotlin_Double_formatToExactDecimals, KDouble value, KInt decimals) {
+  char cstring[16];
```

Comment:
Is 16 indeed always enough?

Reply:
Yes, for large values we use the scientific format.

Reply:
https://stackoverflow.com/questions/9999221/double-precision-decimal-places so considering zero-terminator and dot it seems to be insufficient.

Reply:
This function is only called for values < 1e7 and decimals <= 3

Reply:
I've renamed it to avoid impression of a general purpose number formatting function.

Reply:
Actually it's called on numbers of any magnitude and with any non-negative decimals. I'll think what would be better to do here.

Reply:
I've increased the buffer size to 32 chars: this should be enough with regard to the coming limitations on the function usage:
- decimals will be in range 0..12
- value will be less than 1e+14